packages/helix-shared-config/src/SitemapHandler.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+const fetchAPI = require('@adobe/helix-fetch');
+const { NamedMapHandler } = require('./NamedMapHandler.js');
+
+const { fetch } = process.env.HELIX_FETCH_FORCE_HTTP1
+  ? fetchAPI.context({ alpnProtocols: [fetchAPI.ALPN_HTTP1_1] })
+  /* c8 ignore next */
+  : fetchAPI;
+
+const wrap = (sitemap) => {
+  if (typeof sitemap === 'object') {
+    return {
+      ...sitemap,
+      getXML: async function getXML() {
+        const res = await fetch(new URL(this.destination, this.origin).href);
+        if (!res.ok) {
+          throw new Error(`Failed to fetch ${this.destination}`);
+        }
+        const xml = await res.text();
+        return xml;
+      },
+      getContents: async function getContents() {
+        const xml = await this.getXML();
+        return (xml.match(/<url>(.*?)<\/url>/gs) || []).reduce((acc, url) => {
```

Comment:
would be nicer to extract all urls first....otherwise you do the regexps twice.

Reply:
The structure is

```xml
<url>
  <loc>https://…</loc>
  …
</url>
```

so I first pull all the `<url>` pairs, and then the `<loc>` for each URL.

I'm not clear what you are suggesting.